src/client/Client.cc

```diff
@@ -10512,8 +10512,11 @@ int Client::lazyio_synchronize(int fd, loff_t offset, size_t count)
   Inode *in = f->inode.get();
   
   _fsync(f, true);
-  if (_release(in))
-    check_caps(in, 0);
+  if (_release(in)) {
+    in->auth_cap->wanted |= CEPH_STAT_CAP_SIZE;
```

Comment:
why updating auth cap's wanted

Reply:
I suggested it, as I thought it would inform the MDS that we wanted CEPH_STAT_CAP_SIZE when the check_caps call went out. Maybe that's silly though and we should just call _getattr and assume that the client will eventually get those caps as a matter of course.